external_search.py

```diff
@@ -33,114 +42,158 @@ def __init__(self, url=None, works_index=None):
-                            "filter": {
-                                "en_stop_filter": {
-                                    "type": "stop",
-                                    "stopwords": ["_english_"]
-                                },
-                                "en_stem_filter": {
-                                    "type": "stemmer",
-                                    "name": "english"
-                                },
-                                "en_stem_minimal_filter": {
-                                    "type": "stemmer",
-                                    "name": "english"
-                                },
-                            },
-                            "analyzer" : {
-                                "en_analyzer": {
-                                    "type": "custom",
-                                    "char_filter": ["html_strip"],
-                                    "tokenizer": "standard",
-                                    "filter": ["lowercase", "asciifolding", "en_stop_filter", "en_stem_filter"]
-                                },
-                                "en_minimal_analyzer": {
-                                    "type": "custom",
-                                    "char_filter": ["html_strip"],
-                                    "tokenizer": "standard",
-                                    "filter": ["lowercase", "asciifolding", "en_stop_filter", "en_stem_minimal_filter"]
-                                },
-                            }
-                        }
-                    }
-                },
-            )
-        
-            mapping = {"properties": {}}
-            for field in ["title", "series", "subtitle", "summary", "classifications.term"]:
-                mapping["properties"][field] = {
-                    "type": "string",
-                    "analyzer": "en_analyzer",
-                    "fields": {
-                        "minimal": {
-                            "type": "string",
-                            "analyzer": "en_minimal_analyzer"
-                        }
-                    }
-                }
-            self.indices.put_mapping(
-                doc_type=self.work_document_type,
-                body=mapping,
-                index=self.works_index,
+        if self.indices.exists(index):
+            self.indices.delete(index)
+
+        self.log.info("Creating index %s", index)
+        body = ExternalSearchIndexVersions.latest_body()
+        self.indices.create(index=index, body=body)
+
+    def transfer_current_alias(self, new_index):
+        """Force -current alias onto a new index"""
+        if not self.indices.exists(index=new_index):
+            raise ValueError(
+                "Index '%s' does not exist on this client." % new_index)
+
+        self.works_index = self.__client.works_index = new_index
+        alias_name = self._base_works_index(new_index)+self.CURRENT_ALIAS_SUFFIX
```

Comment:
It shouldn't be possible for the name of the alias to change as part of the transfer. Maybe that means this method should raise an error if the base_works_index differs between the new index and the old index.